pkg/cli/manifest/registermanifest.go

```diff
@@ -38,9 +38,21 @@ func RegisterFile(ctx context.Context, clientFactory *v20231001preview.ClientFac
 		return err
 	}
 
-	logIfEnabled(logger, "Creating resource provider %s", resourceProvider.Name)
+	var locationName string
+	var address string
+
+	if resourceProvider.Location == nil {
+		locationName = v1.LocationGlobal
+	} else {
+		for locationName, address = range resourceProvider.Location {
+			// We support one location per resourceProvider
+			break
+		}
```

Comment:
So, the for loop updates the `locationName` with the first location name from the map... We should add a not here.